server/util/perms/perms.go

```diff
@@ -255,3 +255,31 @@ func AuthenticateSelectedGroupID(ctx context.Context, env environment.Env, proto
 	}
 	return groupID, nil
 }
+
+// AuthenticatedGroupID returns the authenticated group ID from the given
+// context. This is preferred for API requests, since the group ID can be
+// determined directly from the API key. UI requests should instead use
+// `AuthenticateSelectedGroupID`, since the API key is not available, and the
+// user's selected group ID needs to be taken into account.
+//
+// If there is no authenticated user and anonymous usage is enabled, this will
+// return the empty string for the user ID, and a nil error.
+func AuthenticatedGroupID(ctx context.Context, env environment.Env) (string, error) {
```

Comment:
Should we keep this consistent with `AuthenticatedUser` and have it return an error for anonymous users? 
I understand that it makes the API slightly easier to use, but I'm not a fan of the fact that it behaves differently from the other functions in this package.

Reply:
Yeah, it's probably better to have them behave consistently. Done